jwst/ami/bp_fix.py

```diff
@@ -0,0 +1,421 @@
+
+import astropy.io.fits as fits
+import numpy as np
+import logging
+import warnings
+import os
+
+from copy import deepcopy
+from poppy import matrixDFT
+from scipy.ndimage import median_filter
+
+from stdatamodels.jwst.datamodels import dqflags
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+logging.captureWarnings(True)
+
+
+"""pipeline implementation of Jens Kammerer's bp_fix code based on Ireland 2013 algorithm"""
+
+micron = 1.0e-6
+filts = ['F277W', 'F380M', 'F430M', 'F480M', 'F356W', 'F444W']
+# TO DO: get these from some context-controlled place
+filtwl_d = {  # pivot wavelengths
+    'F277W': 2.776e-6,  # less than Nyquist
+    'F380M': 3.828e-6,
+    'F430M': 4.286e-6,
+    'F480M': 4.817e-6,
+    'F356W': 3.595e-6,  # semi-forbidden
+    'F444W': 4.435e-6,  # semi-forbidden
+}
+filthp_d = {  # half power limits
+    'F277W': (2.413e-6, 3.142e-6),
+    'F380M': (3.726e-6, 3.931e-6),
+    'F430M': (4.182e-6, 4.395e-6),
+    'F480M': (4.669e-6, 4.971e-6),
+    'F356W': (3.141e-6, 4.068e-6),
+    'F444W': (3.880e-6, 5.023e-6),
+}
+WL_OVERSIZEFACTOR = 0.1  # increase filter wl support by this amount to 'oversize' in wl space
+
+# GET PUPIL MASK FROM WEBBPSF
+pupil_masks = {}
+webbpsf_path = os.getenv('WEBBPSF_PATH')
+if webbpsf_path is not None:
+    pupilfile_nrm = os.path.join(webbpsf_path,'NIRISS/optics/MASK_NRM.fits.gz')
```

Comment:
Is there a file with this data available through CRDS? If not, can this be added as a new reference file?